### PR TITLE
[REF] Export cleanup - filter at point of query on postal exports

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Follow on from https://github.com/civicrm/civicrm-core/pull/13213/files 
This culls out the non-postal rows at the query rather than deleting them at the end

Before
----------------------------------------
Query fetches rows that can't be emailed and THEN they are culled

After
----------------------------------------
Where filters added to query

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by mattwire** [comment]
See https://github.com/mattwire/civicrm-core/commit/4b8a3a2229192458dad3024bb93c1da2d97f0b25 for some tweaks to fix issues in #13213 and get merge household working

File: CRM/Export/BAO/ExportProcessor.php [hunk 1]
```diff
@@ -428,6 +458,23 @@ public function getHeaderForRow($field) {
...
+    if ($this->isPostalableOnly) {
+      if (array_key_exists('street_address', $returnProperties)) {
+        $addressWhere = " civicrm_address.street_address <> ''";
+        if (array_key_exists('supplemental_address_1', $returnProperties)) {
+          // We need this to be an OR rather than AND on the street_address so, hack it in.
+          $addressOptions = CRM_Core_BAO_Setting::valueOptions(CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME,
+            'address_options', TRUE, NULL, TRUE
+          );
+          if (!empty($addressOptions['supplemental_address_1'])) {
+            $addressWhere .= " AND civicrm_address.supplemental_address_1 <> ''";
```
Comment (mattwire) [inline CRM/Export/BAO/ExportProcessor.php]: @eileenmcnaughton I think this should be OR not AND!

Comment (eileenmcnaughton) [reply CRM/Export/BAO/ExportProcessor.php]: I think you are right - if street address is not empty or supplemental address is not empty then we have an address we can email - updated

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
@eileenmcnaughton This is tested and working now on a live site. Ok to merge.

I did have to make one change to set the table type to MYISAM instead of INNODB here https://github.com/eileenmcnaughton/civicrm-core/blob/30eab4c17d15d688acc039a5974a695386271a9e/CRM/Export/BAO/Export.php#L713 as I get row size to large otherwise.  But that's a long-standing issue before/after this PR :-)  It does, however, allow me to test and verify that all three contact export modes with/without postal selected now work correctly! Well done!

**Pull request merged**